CMakePresets.json

```diff
@@ -1,141 +1,174 @@
-        "LADYBIRD_SOURCE_DIR": "${fileDir}"
-      }
-    },
-    {
-      "name": "Sanitizer",
-      "inherits": "default",
-      "configurePreset": "Sanitizer",
-      "environment": {
-        "ASAN_OPTIONS": "strict_string_checks=1:check_initialization_order=1:strict_init_order=1:detect_stack_use_after_return=1:allocator_may_return_null=1",
-        "UBSAN_OPTIONS": "print_stacktrace=1:print_summary=1:halt_on_error=1"
-      }
-    },
-    {
-      "name": "Debug",
-      "inherits": "default",
-      "configurePreset": "Debug"
-    }
-  ]
-}
+    "configurePresets": [
+        {
+            "name": "default",
+            "displayName": "Default Config",
+            "description": "Default build using Ninja generator",
+            "generator": "Ninja",
+            "binaryDir": "${fileDir}/Build/ladybird",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+                "SERENITY_CACHE_DIR": "${fileDir}/Build/caches",
+                "CMAKE_TOOLCHAIN_FILE": "${fileDir}/Toolchain/Tarballs/vcpkg/scripts/buildsystems/vcpkg.cmake",
+                "VCPKG_INSTALL_OPTIONS": "--no-print-usage",
+                "VCPKG_OVERLAY_TRIPLETS": "${fileDir}/Meta/CMake/vcpkg/release-triplets"
+            },
+            "environment": {
+                "LADYBIRD_SOURCE_DIR": "${fileDir}",
+                "VCPKG_ROOT": "${fileDir}/Toolchain/Tarballs/vcpkg",
+                "VCPKG_BINARY_SOURCES": "clear;files,${fileDir}/Toolchain/Build/vcpkg-binary-cache,readwrite"
+            },
+            "vendor": {
+                "jetbrains.com/clion": {
+                    "toolchain": "Default"
+                }
+            }
+        },
+        {
+            "name": "Debug",
+            "inherits": "default",
+            "displayName": "Debug Config",
+            "description": "Debug build using Ninja generator",
+            "binaryDir": "${fileDir}/Build/ladybird-debug",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Debug",
+                "VCPKG_OVERLAY_TRIPLETS": "${fileDir}/Meta/CMake/vcpkg/debug-triplets"
+            }
+        },
+        {
+            "name": "Sanitizer",
+            "inherits": "Debug",
+            "displayName": "Sanitizer Config",
+            "description": "Debug build using Sanitizers",
+            "binaryDir": "${fileDir}/Build/ladybird-sanitizers",
+            "cacheVariables": {
+                "ENABLE_UNDEFINED_SANITIZER": "ON",
+                "ENABLE_ADDRESS_SANITIZER": "ON",
+                "VCPKG_OVERLAY_TRIPLETS": "${fileDir}/Meta/CMake/vcpkg/sanitizer-triplets"
+            }
+        },
+        {
+            "hidden": true,
+            "name": "vcpkg_ci",
+            "description": "Use the GitHub Actions vcpkg cache",
+            "environment": {
+                "VCPKG_BINARY_SOURCES": "clear;x-gha,readwrite"
+            }
+        },
+        {
+            "name": "CI",
+            "inherits": [
```

Comment:
please configure your editor to not replace \n with \r\n. These files were checked in initially with unix line endings, and they should stay that way.